doc/notebooks/parallelization_tutorial.ipynb

```diff
@@ -0,0 +1,510 @@
+    "Besides individual tools, the `--exec` interface can execute an entire script to enable more complex workflows."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d40d3fd8-be14-46fa-bb39-456cb140b3ae",
+   "metadata": {},
+   "source": [
+    "#### Safely modifying computational region\n",
+    "\n",
+    "Sometimes modifying computational region in a script is needed. It is a good practice to not change the global computational region, which effectively modifies a file in a mapset,\n",
+    "but only change the environment variable `GRASS_REGION`.\n",
+    "Here, we modified the previous viewshed example to compute in parallel viewsheds with different extents:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0366af36-0556-4be0-bf08-5a813942c1bb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile example.py\n",
+    "import os\n",
+    "from multiprocessing import Pool\n",
+    "import grass.script as gs\n",
+    "\n",
+    "def viewshed(point):\n",
+    "    x, y, cat = point\n",
+    "    # copy current environment, modify and pass it to r.viewshed\n",
+    "    env = os.environ.copy()\n",
+    "    env[\"GRASS_REGION\"] = gs.region_env(e=x + 300, w=x - 300, n=y + 300, s=y - 300, align=\"elevation\")\n",
+    "    gs.run_command(\"r.viewshed\", input=\"elevation\", output=f\"viewshed_{cat}\", coordinates=(x, y), max_distance=300, env=env)\n",
+    "    return f\"viewshed_{cat}\"\n",
+    "\n",
+    "if __name__ == \"__main__\":\n",
+    "    viewpoints = [(633709, 225663, 1),\n",
+    "                  (639432, 222826, 2),\n",
+    "                  (640385, 220502, 3),\n",
+    "                  (636521, 219353, 4)]\n",
+    "    with Pool(processes=4) as pool:\n",
+    "        maps = pool.map(viewshed, viewpoints)\n",
+    "    print(maps)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bc310e4d-5b40-4bd4-a579-892314fe5e99",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%run example.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c1cd8369-d1e6-471e-b4e1-88b0cbdf57da",
+   "metadata": {},
+   "source": [
+    "#### Safely modifying vectors with attributes in a single mapset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "28b0c12c-a076-460f-b93b-764c1907c96f",
+   "metadata": {},
+   "source": [
+    "By default vector maps share a single sqlite database file, however sqlite does not support concurrent write access. That poses a problem when modifying vectors with attributes in parallel. While this can be solved by running the computations in separate mapsets, it is also possible to change the default behavior to write attributes of each vector to the vector's individual sqlite file. This behavior can be activated after a new mapset is created with:\n",
+    "\n",
+    "```\n",
+    " db.connect driver=sqlite database='$GISDBASE/$LOCATION_NAME/$MAPSET/vector/$MAP/sqlite.db'\n",
+    "```"
+   ]
```

Comment:
Alternatively, a PostgreSQL or another database backend can be used for attributes to offload the parallel writing to the database system.